### PR TITLE
Whitelist regexp_parser warning and bump regexp_parser to 0.4.9

### PR DESCRIPTION
**Pull request by zaidan**

### CODE REVIEW

File: mutant.gemspec [hunk 1]
```diff
@@ -35,7 +35,7 @@ Gem::Specification.new do |gem|
   gem.add_runtime_dependency('equalizer',     '~> 0.0.9')
   gem.add_runtime_dependency('anima',         '~> 0.3.0')
   gem.add_runtime_dependency('concord',       '~> 0.1.5')
-  gem.add_runtime_dependency('regexp_parser', '~> 0.4.4')
+  gem.add_runtime_dependency('regexp_parser', '~> 0.4.5', '< 0.4.6')
```
Comment (mbj) [inline mutant.gemspec]: I suspect specifying `~> 0.4.6` just had the same effect?

Comment (zaidan) [reply mutant.gemspec]: According to bundler documentation `~> 0.4.6` would be identical to `>= 0.4.6 and < 0.5`. `0.4.6` is not working. If they had use semantic versioning from the beginning, we could write `~> 0.4.5`. This is a stupid Ruby issue. I was wondering if it is possible to switch to 4 version number segments, like in Haskell.

Comment (mbj) [reply mutant.gemspec]: yeah, I forgot. Do not do much ruby anymore.

So why do we do not want `0.4.6`?

Comment (zaidan) [reply mutant.gemspec]:
```bash
Mutant on ruby corpus rubyspec does not fail on mutation generation
     Failure/Error: fail RegistryError, "No entry for: #{type.inspect}"
     
     Mutant::Registry::RegistryError:
       No entry for: :regexp_xgrapheme_type
```
see:
* https://circleci.com/gh/zaidan/mutant/19
* https://circleci.com/gh/zaidan/mutant/18

Comment (mbj) [reply mutant.gemspec]: So instead of keeping mutant on an older regexp parser, why not simply add these new (unmutated) nodes to the noop handler?

Comment (zaidan) [reply mutant.gemspec]: Do you mean adding :regexp_xgrapheme_type to `handle(...)` in `lib/mutant/mutator/node/noop.rb`? This produces the following error:
```bash
register': Invalid type registration: :regexp_xgrapheme_type (Mutant::Registry::RegistryError)
```

Comment (mbj) [reply mutant.gemspec]: there is a regexp mutator level noop.

Comment (zaidan) [reply mutant.gemspec]: do you mean some where in `lib/mutant/mutator/node/regexp/**`? There is no file called `noop`. I did a grep for `noop` and could not find something useful.

Comment (mbj) [reply mutant.gemspec]: There is a list of nodes we do not explicitly mutate inside that namespace. I do, right now, not have the time to point you to it, sorry. Busy with commercial clients.

Comment (zaidan) [reply mutant.gemspec]: ok, fixed this and another issue with `>= 0.4.7`. Got it working with `~> 0.4.9`.

Comment (mbj) [reply mutant.gemspec]: good, this is the much better change.

**Comment by zaidan** [comment]
bump

**Pull request merged**